frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/PeriodsAgoOptionComponent.tsx

```diff
@@ -1,151 +1,95 @@
-import { useCallback, useMemo, useRef, useState } from "react";
-import type {
-  Dispatch,
-  KeyboardEvent,
-  MouseEvent,
-  SetStateAction,
-} from "react";
+import { useCallback, useRef } from "react";
+import type { KeyboardEvent, MouseEvent } from "react";
 import { Group, Text, Box } from "metabase/ui";
-import type { SelectedComparisonPeriodsAgo } from "metabase-types/api";
+import type { SmartScalarComparisonPeriodsAgo } from "metabase-types/api";
+import type { COMPARISON_TYPES } from "metabase/visualizations/visualizations/SmartScalar/constants";
 import { NumberInputStyled } from "./PeriodsAgoOptionComponent.styled";
 import { MenuItemStyled } from "./MenuItem.styled";
+import type { HandleEditedValueChangeType } from "./SmartScalarSettingsWidgets";
 
-type PeriodsAgoOptionComponentProps = {
-  isSelected: boolean;
+type PeriodsAgoMenuOptionProps = {
+  "aria-selected": boolean;
+  editedValue?: SmartScalarComparisonPeriodsAgo;
```

Comment:
In response to figuring out a more elegant solution for handling `periodsAgo` value submission on menu closing, this was introduced. It might help to review just the file itself since most of the diff is removing the hacks.